### PR TITLE
Upgrade to Rails 6.1

### PR DESCRIPTION
**Pull request by jcoyne**
Note, a good deal of this was necessary due to a regression in Rails 6.1: https://github.com/rails/rails/issues/41010

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -508,9 +506,8 @@ GEM
     zeitwerk (2.4.2)
 
 PLATFORMS
-  ruby
```
Comment (mjgiarlo) [inline Gemfile.lock:511]: At some point, maybe we should have a conversation about the PLATFORMS values and whether it's OK for them to keep changing.

File: Gemfile.lock [hunk 2]
```diff
@@ -508,9 +506,8 @@ GEM
...
   x86_64-darwin-19
-  x86_64-linux
+  x86_64-linux-musl
```
Comment (mjgiarlo) [inline Gemfile.lock:510]: I pulled the branch and this didn't cause me any problems FWIW.

File: app/components/collections/workflow_review_component.rb [hunk 3]
```diff
@@ -19,7 +19,7 @@ def review_workflow_status
 
     sig { returns(T.nilable(String)) }
     def reviewers
-      collection.reviewers.map(&:sunetid).join(', ')
+      collection.reviewed_by.map(&:sunetid).join(', ')
```
Comment (mjgiarlo) [inline app/components/collections/workflow_review_component.rb:22]: Can you say more about this change?

File: spec/forms/work_form_spec.rb [hunk 4]
```diff
@@ -78,73 +78,87 @@
...
 
       it 'validates with contributors in place' do
-        expect(form.errors.messages).not_to include(contributor_error)
+        expect(errors).to be_empty
       end
     end
   end
 
   describe 'email validation' do
+    let(:errors) { form.errors.where(:contact_email) }
```
Comment (mjgiarlo) [inline spec/forms/work_form_spec.rb:90]: Heads-up, @jermnelson -- this may trip up your PR if this one is merged first.

**Review by mjgiarlo — COMMENTED** [review]
Have a few small questions.

File: app/components/collections/workflow_review_component.rb (hunk 3 above)
Comment (jcoyne) [reply app/components/collections/workflow_review_component.rb:22]: Sure: https://github.com/rails/rails/issues/41010

File: Gemfile.lock (hunk 1 above)
Comment (jcoyne) [reply Gemfile.lock:511]: I never look at this, I just run `bundle install`

**Review by mjgiarlo — APPROVED** [review]
Feel free to merge once :green_heart:

**Pull request merged**